app/actions/websocket/posts.ts

```diff
@@ -0,0 +1,191 @@
+    // const collapsedThreadsEnabled = isCollapsedThreadsEnabled(state);
+    // actions.push(receivedNewPost(post, collapsedThreadsEnabled));
+    if (!shouldIgnorePost(post)) {
+        let markAsRead = false;
+        let markAsReadOnServer = false;
+
+        if (!myChannel?.manuallyUnread) {
+            if (
+                post.user_id === currentUserId &&
+                !isSystemMessage(post) &&
+                !isFromWebhook(post)
+            ) {
+                markAsRead = true;
+                markAsReadOnServer = false;
+            } else if ((post.channel_id === currentChannelId)) { // TODO: THREADS && !viewingGlobalThreads) {
+                // Don't mark as read if we're in global threads screen
+                // the currentChannelId still refers to previously viewed channel
+                markAsRead = true;
+                markAsReadOnServer = true;
+            }
+        }
+
+        if (markAsRead && markAsReadOnServer) {
+            markChannelAsRead(serverUrl, post.channel_id);
+        } else if (markAsRead) {
+            const viewedAt = await markChannelAsViewed(serverUrl, post.channel_id, true);
+            if (viewedAt instanceof Model) {
+                models.push(viewedAt);
+            }
+        } else {
+            const unreadAt = markChannelAsUnread(serverUrl, post.channel_id, 1, msg.data.mentions, false, true);
+            if (unreadAt instanceof Model) {
+                models.push(unreadAt);
+            }
+        }
+    }
+
+    database.operator.batchRecords(models);
+}
+
+export async function handlePostEdited(serverUrl: string, msg: WebSocketMessage) {
+    const database = DatabaseManager.serverDatabases[serverUrl];
+    if (!database) {
+        return;
+    }
+    const currentUserId = await queryCurrentUserId(database.database);
+    const data = JSON.parse(msg.data.post);
+    const post = {
+        ...data,
+        ownPost: data.user_id === currentUserId,
+    };
+
+    try {
+        fetchPostAuthors(serverUrl, [post]);
+    } catch {
+        // Do nothing
+    }
+
+    database.operator.handlePosts({
+        actionType: ActionType.POSTS.RECEIVED_NEW,
+        order: [post.id],
+        posts: [post],
+    });
+}
+
+export function handlePostDeleted(serverUrl: string, msg: WebSocketMessage) {
+    const data = JSON.parse(msg.data.post);
+
+    removePostById(serverUrl, data.id);
+}
+
+export async function handlePostUnread(serverUrl: string, msg: WebSocketMessage) {
+    const database = DatabaseManager.serverDatabases[serverUrl];
+    if (!database) {
+        return;
+    }
+
+    const myChannel = await queryMyChannel(database.database, msg.broadcast.channel_id);
+
+    if (!myChannel?.manuallyUnread) {
+        const delta = myChannel ? myChannel.messageCount - msg.data.msg_count : msg.data.msg_count;
+        markChannelAsUnread(serverUrl, msg.broadcast.channel_id, delta, msg.data.mention_count, true);
```

Comment:
this is preparing the records and they are never committed to the db

Reply:
The true is for the `manuallyUnread`. So `markChannelAsUnread` is receiving a false for prepareRecordsOnly, and it is updating the database.